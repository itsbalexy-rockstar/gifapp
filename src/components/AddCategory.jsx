import React from 'react';
import { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types'

const AddCategory = ({setCategories}) => {
  
    const [inputValue, setInputValue] = useState('');
    const handleInputValue = (e) => {
        setInputValue(e.target.value)
    }
    const handleSubmit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length > 2){
            setCategories((categories) => [inputValue, ...categories])
            setInputValue('')
        }
    }

    return (
      <AddCategoryStyled>
          <form action="" onSubmit={handleSubmit}>
            <input 
            placeholder='Search'
            type="text" 
            value={inputValue}
            onChange={handleInputValue}
            id='categoryValue'
            />
          </form>
      </AddCategoryStyled>
  );
};

const AddCategoryStyled = styled.form`
    display: flex;
    justify-content: center;
    position: relative;
    overflow: hidden;
    width: 300px;
    input{
        width: 100%;
        background-color: transparent;
        color: #eeeeee;
        border: none;
        border-bottom: 1px solid #eeeeee;
        outline: 0px;
        font-size: 1rem;
        padding: .4rem;
    }
`
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
