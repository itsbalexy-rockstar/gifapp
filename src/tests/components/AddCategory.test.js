import React from 'react';
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";
import '@testing-library/jest-dom'

describe('Tests in <AddCategory />', () => {
    
    const setCategories = jest.fn()
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>)
    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory setCategories={setCategories}/>)
    })

    test('Should show the correctly component', () => {
      expect(wrapper).toMatchSnapshot()
    });
    test('Should change the input box', () => {
        const input = wrapper.find('input')
        const value = 'Hola'

        input.simulate('change', {target: {value}})
    });
    test('Should not continue calling the function', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}})
        expect(setCategories).not.toHaveBeenCalled()
    });
    test('Should call setCategories function and cleaning the input box', () => {
        const value = 'Holi'
        const inputValue = wrapper.find('input').prop('value')
        wrapper.find('input').simulate('change', {target: {value}})
        wrapper.find('form').simulate('submit', {preventDefault(){}})
        expect(setCategories).toHaveBeenCalledTimes(1)
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) )
        expect(inputValue).toBe('')
    });

})