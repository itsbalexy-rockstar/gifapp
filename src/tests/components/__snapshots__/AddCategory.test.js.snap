// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AddCategory /> Should show the correctly component 1`] = `
<styled.form>
  <form
    action=""
    onSubmit={[Function]}
  >
    <input
      id="categoryValue"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </form>
</styled.form>
`;
